src/provider/HttpFile/ConfigurationServiceHttpFile.ts

```diff
@@ -22,16 +22,32 @@ import {
   getRepositoryKeyNotFoundErrorMessage
 } from '../../utils';
 
+interface ConfigurationServiceHttpFileRequest {
+  token: string,
+  disableCache?: boolean
+}
+
 export class ConfigurationServiceHttpFile implements ConfigurationService {
-  constructor(private token: string) {
+  private token: string = '';
+  private disableCache?: boolean;
+
+  constructor(request: string | ConfigurationServiceHttpFileRequest) {
+    if(typeof request === 'string') {
```

Comment:
Please add a comment that     if(typeof request === 'string') { is for backward compatibility and it's deprecated

Reply:
done